File: fixtures/hub-gold-rails-routes/app/controllers/items_controller.rb
# frozen_string_literal: true
# Items controller for hub-gold-rails-routes (G10115 / D6540).

class ItemsController < ActionController::API
  def index
    render json: true
  end

  def show
    render json: { id: params[:id] }
  end

  def create
    render json: { created: true }, status: 201
  end

  def update
    render json: { updated: true, id: params[:id] }
  end

  def destroy
    render json: true
  end

  def patch
    render json: { patched: true, id: params[:id] }
  end
end

File: fixtures/hub-gold-phoenix-controllers/lib/hub_gold_web/controllers/items_controller.ex
defmodule HubGoldWeb.ItemsController do
  use Phoenix.Controller

  # Items controller for hub-gold-phoenix-controllers (G10126 / D6540).

  def index(conn, _params) do
    json(conn, true)
  end

  def show(conn, params) do
    json(conn, %{id: params["id"]})
  end

  def create(conn, _params) do
    conn
    |> put_status(201)
    |> json(%{created: true})
  end

  def update(conn, params) do
    json(conn, %{updated: true, id: params["id"]})
  end

  def destroy(conn, _params) do
    json(conn, true)
  end

  def patch(conn, params) do
    json(conn, %{patched: true, id: params["id"]})
  end
end
